### PR TITLE
TT-8308 Remove some things to decrease CVE surface in plugin compiler

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

https://tyktech.atlassian.net/browse/TT-8308
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit:  
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4489160851)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: e1c7d8df7ea022d16e124624405c8b8d229345e1 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4489252657)

File: ci/images/plugin-compiler/Dockerfile [hunk 1]
```diff
@@ -16,6 +16,45 @@ RUN mkdir -p  $TYK_GW_PATH $PLUGIN_SOURCE_PATH
 COPY ci/images/plugin-compiler/data/build.sh /build.sh
 RUN chmod +x /build.sh
 
+RUN echo "Starting image build for $(grep PRETTY_NAME /etc/os-release)" \
```
Comment (alephnull) [inline ci/images/plugin-compiler/Dockerfile:19]: No need to install.

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: 90a2f774f68d5779c5af8b5c1a5ad5c9021da3e4 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4489338598)

File: ci/images/plugin-compiler/Dockerfile [hunk 2]
```diff
@@ -16,6 +16,45 @@ RUN mkdir -p  $TYK_GW_PATH $PLUGIN_SOURCE_PATH
...
+        openssl                                        \
+        libssl-dev                                     \
+        unzip                                          \
+        sudo                                           \
+        jq                                             \
+        rpm                                            \
+        ruby-dev
+
+# Remove some things to decrease CVE surface
+RUN  apt-get remove -y --allow-remove-essential --auto-remove curl\
```
Comment (alephnull) [inline ci/images/plugin-compiler/Dockerfile:55]: Remove git and the other package here?

**Review by alephnull — CHANGES_REQUESTED** [review]

File: ci/images/plugin-compiler/Dockerfile (hunk 2 above)
Comment (tbuchaillot) [reply ci/images/plugin-compiler/Dockerfile:20]: git is needed after for custom repos in plugins - let's just remove curl for now.

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: 420e98ca6d9896a3415e7d5037628e787f8fce25 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4490968903)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: 2cec0f4f33bdb3932afa821828871e51855d9037 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4491217752)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: cf468517c26f1c4ee9e8af5f7e7c33a6c1fba6c5 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4498677002)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: 6b5ba8fde92d2557f30da77f5ece3347bb40c043 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4499090394)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: a11aa32f8caf39ce54fb4df0d39d5af0b7f49ba2 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4499191515)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4889/merge
Commit: d871414c122061646ba6b97dcfa090b40467f043 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/4499435528)

**Pull request closed without merging**